src/Microsoft.AspNet.Server.Kestrel/Http/SocketOutput.cs

```diff
@@ -388,6 +387,22 @@ void ISocketOutput.Write(ArraySegment<byte> buffer, bool immediate, bool chunk)
 
         Task ISocketOutput.WriteAsync(ArraySegment<byte> buffer, bool immediate, bool chunk, CancellationToken cancellationToken)
         {
+            if (cancellationToken.IsCancellationRequested)
```

Comment:
We want to register with the cancellation token for writes that actually result in an incomplete task being returned.